### PR TITLE
Access requests accepted/declined requests give a 404 page

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ctrl-alt-soneca** [comment]
Requestor:
email accept

Owner:

File: zenodo_accessrequests/views/requests.py [hunk 1]
```diff
@@ -156,7 +156,8 @@ def confirm(pid, record, template, **kwargs):
 
     # Confirm email address.
     if r.status != RequestStatus.EMAIL_VALIDATION:
-        abort(404)
+        error_message = "The status of your access request is: " + r.status.value + "."
```
Comment (slint) [inline zenodo_accessrequests/views/requests.py:159]: - There should be some follow-up information suggested to the user, so they can know what to expect. I.e. if the request is `RequestStatus.PENDING` it should say something like 'Your email address has been verified. and the access request is now pending a decision from the record owner.'
- If it's accepted or rejected, it should say "The request has already been {status}. Check your mail inbox for the record owner's response."
- Also formatting style (see comment/suggestion below)

File: zenodo_accessrequests/views/requests.py [hunk 2]
```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
 
     # Confirm email address.
     if r.status != RequestStatus.EMAIL_VALIDATION:
-        abort(404)
+        if(str(r.status.value).lower() == "pending"):
```
Comment (slint) [inline zenodo_accessrequests/views/requests.py:159]:
```suggestion
        if r.status == RequestStatus.PENDING:
```

- No need for parentheses around `if ...` statements :)
- It's better to compare with the enum value directly

File: zenodo_accessrequests/views/requests.py [hunk 3]
```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
...
+                    "Your email address has been verified.The access request "
+                    "is now pending a decision from the record owner."
+                )
+            )
+        else:
+            abort(
+                400, 
+                (
+                    "The request has already been {}."
+                    " Check your mail inbox for the record owner's response."
```
Comment (slint) [inline zenodo_accessrequests/views/requests.py:172]:
```suggestion
                    "The request has already been {}. "
                    "Check your mail inbox for the record owner's response."
```

I have a slight preference for spaces to be at the end of the line for multiline strings.

File: zenodo_accessrequests/views/requests.py [hunk 4]
```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
 
     # Confirm email address.
     if r.status != RequestStatus.EMAIL_VALIDATION:
-        abort(404)
+        if(str(r.status.value).lower() == "pending"):
+            abort(
+                400, 
+                (
+                    "Your email address has been verified.The access request "
```
Comment (slint) [inline zenodo_accessrequests/views/requests.py:163]:
```suggestion
                    "Your email address has been verified. The access request "
```

File: zenodo_accessrequests/views/settings.py [hunk 5]
```diff
@@ -134,9 +134,19 @@ def index():
 def accessrequest(request_id):
     """Accept/reject access request."""
     r = AccessRequest.get_by_receiver(request_id, current_user)
-    if not r or r.status != RequestStatus.PENDING:
+    if not r:
         abort(404)
 
+    if(r.status != RequestStatus.PENDING):
```
Comment (slint) [inline zenodo_accessrequests/views/settings.py:140]:
```suggestion
    if r.status != RequestStatus.PENDING:
```

**Review by slint — COMMENTED** [review]
Last changes and it can go in. I might need to add GH Actions CI/CD in this repo, since now there are no tests running...

File: zenodo_accessrequests/views/requests.py [hunk 6]
```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
...
+                (
+                    "Your email address has been verified. The access request "
+                    "is now pending a decision from the record owner."
+                )
+            )
+        else:
+            abort(
+                400, 
+                (
+                    "The request has already been {}. "
```
Comment (slint) [inline zenodo_accessrequests/views/requests.py:171]:
```suggestion
                    "The request has been {}. "
```

File: zenodo_accessrequests/views/settings.py [hunk 7]
```diff
@@ -134,9 +134,19 @@ def index():
...
-    if not r or r.status != RequestStatus.PENDING:
+    if not r:
         abort(404)
 
+    if r.status != RequestStatus.PENDING:
+        abort(
+            400, 
+            (
+                "This link is not available anymore since the access "
+                "request has been {} already. You can navigate to your "
```
Comment (slint) [inline zenodo_accessrequests/views/settings.py:145]:
```suggestion
                "request has been {}. You can navigate to your "
```

**Comment by slint** [comment]
Closed in favor of #38

**Pull request closed without merging**